### PR TITLE
Add support for SoC separate var partition

### PR DESCRIPTION
**Pull request by Yarboa**
Updating tests/e2e/set-ffi-env-e2e to mkfs.ext4
On free sde partiton it finds VROOM-21987

### CODE REVIEW

**Comment by dougsland** [comment]
I don't believe these are related to your change, already exists as we talked:

/tests/ffi/memory (timeout)
```
Copying config sha256:728d9197c2ef33c51583e7e162d6055d654208c789a54dd033cb82607f81b444
Writing manifest to image destination
Getting image source signatures
Writing manifest to image destination

Maximum test time '20m' exceeded.
Adjust the test 'duration' attribute if necessary.
https://tmt.readthedocs.io/en/stable/spec/tests.html#duration
```

/tests/ffi/qm-oom-score-adj
```
Error: OCI runtime error: crun: the requested cgroup controller `pids` is not available
[ INFO  ] PASS: qm.container oom_score_adj value == 500
./test.sh: line 39: [: cat: /proc/0/oom_score_adj: No such file or directory: integer expression expected
[ INFO  ] FAIL: qm containers oom_score_adj != 750. Current value is cat: /proc/0/oom_score_adj: No such file or directory
Shared connection to 13.59.78.82 closed.
```

**Comment by dougsland** [comment]
Output from CI/CD also from your patch (pids exists from cgroup2):
```
Run echo "Checking cgroup version and settings..."
Checking cgroup version and settings...
nodev	cgroup
nodev	cgroup2
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
#subsys_name	hierarchy	num_cgroups	enabled
cpuset	0	[1](https://github.com/containers/qm/actions/runs/10588279250/job/29340520739?pr=518#step:3:1)09	1
cpu	0	109	1
cpuacct	0	109	1
blkio	0	109	1
memory	0	109	1
devices	0	109	1
freezer	0	109	1
net_cls	0	10[9](https://github.com/containers/qm/actions/runs/10588279250/job/29340520739?pr=518#step:3:10)	1
perf_event	0	[10](https://github.com/containers/qm/actions/runs/10588279250/job/29340520739?pr=518#step:3:11)9	1
net_prio	0	109	1
hugetlb	0	109	1
pids	0	109	1
rdma	0	109	1
misc	0	109	1
```

File: tests/e2e/lib/diskutils [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+  select_disk_to_partition
+
+  local slash_var
+  slash_var="/var"
+
+  # Check is SoC aboot exist, else regular image
+  # VM use case
+  if [ -e /sys/devices/soc0/machine ]; then
+        if grep -qi SA8775P /sys/devices/soc0/machine; then
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:80]: SA8775P seems a magic name, can we have a constant to describe it?

File: tests/e2e/lib/diskutils [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          mkfs.ext4 "/dev/${DISK}${PART_ID}"
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:81]: For most of the commands in this function we need if_error_exit or bash -ex in the header of the file. Its your personal preference. I prefer if_error_exit so it's less verbose and I can validate every error.

File: tests/e2e/lib/diskutils [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+         DISK="${disk}"
+         if [[ $(echo "${disk}" | grep -c nvme) -eq 1 ]]; then
+           part_id=p${part_id}
+           PART_ID="${part_id}"
+           info_message "select_disk_to_partition, found /dev/${DISK}${PART_ID}"
+           info_message "=============================="
+           break;
+         fi
+      fi
+      if [ -e /sys/devices/soc0/machine ]; then
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:49]: will be always soc0 and sde?

Comment (dougsland) [reply tests/e2e/lib/diskutils:49]: I believe it's possible to detect such machine via cat /etc/machine-id OR sudo dmidecode -s system-uuid

File: tests/e2e/lib/diskutils (hunk 1 above)
Comment (Yarboa) [reply tests/e2e/lib/diskutils:80]: OK

File: tests/e2e/lib/diskutils (hunk 2 above)
Comment (Yarboa) [reply tests/e2e/lib/diskutils:81]: OK

File: tests/e2e/lib/diskutils (hunk 3 above)
Comment (Yarboa) [reply tests/e2e/lib/diskutils:49]: not in the RideSXchip
 dmidecode -s system-uuid return empty result
 [root@localhost ~]# dmidecode 
# dmidecode 3.6
# No SMBIOS nor DMI entry point found, sorry.

Booted by android,
I will ignore this one

Comment (dougsland) [reply tests/e2e/lib/diskutils:48]: last question before the merge @Yarboa, whats the output of:
```
sudo grep -i "ridesx4" /sys/class/dmi/id/*

```

Comment (Yarboa) [reply tests/e2e/lib/diskutils:48]:
```
 grep -ri "ridesx4" /sys/class/dmi/id/*
grep: /sys/class/dmi/id/*: No such file or directory

```

but you have this 
```
cat /etc/machine-id 
506b9b0ac0c6476ab57bc5c553cf4a4a

```


I suspect it is related to aboot

Comment (dougsland) [reply tests/e2e/lib/diskutils:48]: /etc/machine-id should be unique per machine, do not help us in this case.

File: tests/e2e/lib/diskutils [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+         if [[ $(echo "${DISK}" | grep -c nvme) -eq 1 ]]; then
+           PART_ID="p${part_id}"
+           info_message "select_disk_to_partition, found /dev/${DISK}${PART_ID}"
+           info_message "=============================="
+           break;
+         fi
+      fi
+      if [ -e /sys/devices/soc0/machine ]; then
+        if grep -qi "${QC_SOC}" /sys/devices/soc0/machine; then
+           if [[ ${disk} == "sde" ]]; then
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:50]: sde should be a constant.

File: tests/e2e/lib/diskutils [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  ###########################################################################
+
+  select_disk_to_partition
+
+  local slash_var
+  slash_var="/var"
+
+  # Check is SoC aboot exist, else regular image
+  if [ -e /sys/devices/soc0/machine ]; then
+        if grep -qi "${QC_SOC}" /sys/devices/soc0/machine; then
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:78]: /sys/devices/soc0/machine should be a constant as multiple places use this string.

File: tests/e2e/lib/diskutils [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  rsync -aqxXP /var/* /new_var
+  if_error_exit "Error: rsync failed"
+  systemctl stop var-lib-nfs-rpc_pipefs.mount
+  umount /new_var
+  rm -rf "${slash_var:?}"/*
+  info_message "Create_qm_disks, prepare and mount ${slash_var}"
+  info_message "=============================="
+  mount "/dev/${DISK}${PART_ID}" "${slash_var}"
+  if_error_exit "Error: mount /dev/${DISK}${PART_ID} on ${slash_var} failed"
+  systemctl start var-lib-nfs-rpc_pipefs.mount
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:104]: should we check for error in this one too ?

File: tests/e2e/set-ffi-env-e2e [hunk 7]
```diff
@@ -273,12 +242,27 @@ then
...
+    create_qm_var_part
+fi
+
 echo
 info_message "Checking if QM already installed"
 info_message "=============================="
 QM_STATUS="$(systemctl is-enabled qm 2>&1)"
 if [ "$QM_STATUS" == "generated" ]; then
    if [ "$(systemctl is-active qm)" == "active" ]; then
+       systemctl restart qm
```
Comment (dougsland) [inline tests/e2e/set-ffi-env-e2e:265]: This is interesting: if it's active why restart?

File: tests/e2e/lib/diskutils (hunk 4 above)
Comment (Yarboa) [reply tests/e2e/lib/diskutils:50]: Ack

File: tests/e2e/lib/diskutils (hunk 5 above)
Comment (Yarboa) [reply tests/e2e/lib/diskutils:78]: Ack

File: tests/e2e/set-ffi-env-e2e (hunk 7 above)
Comment (Yarboa) [reply tests/e2e/set-ffi-env-e2e:265]: doing restart for the new /var partition added earlier.
I can check for disk partition

File: tests/e2e/lib/diskutils (hunk 6 above)
Comment (Yarboa) [reply tests/e2e/lib/diskutils:104]: Ack

File: tests/e2e/lib/diskutils [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  local disk_table
+  disk_table=$(lsblk --noheadings --raw)
+  local disks_arr
+  disks_arr=$(echo "$disk_table" | awk '$1~// && $6=="disk" {print $1}')
+  info_message "select_disk_to_partition, found ${disks_arr}"
+  info_message "=============================="
+
+  for disk in $disks_arr; do
+     if [[ ${disk} == "vda" || \
+           $(echo "${disk_table}" | grep -c "${disk}" ) -eq 1 && ${disk} != "zram0" ]];then
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:37]: looks like out of indent

File: tests/e2e/lib/diskutils [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  # VM use case
+       mkfs.xfs "/dev/${DISK}${PART_ID}"
+       if_error_exit "Error: mkfs.xfs /dev/${DISK}${PART_ID} failed on VM"
+       info_message "Create_qm_disks, prepare regular xfs fs"
+       info_message "=============================="
+
+  fi
+
+  mkdir -p /new_var
+  mount "/dev/${DISK}${PART_ID}" /new_var
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:94]: if_mount_error?

File: tests/e2e/lib/diskutils [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  rsync -aqxXP /var/* /new_var
+  if_error_exit "Error: rsync failed"
+  systemctl stop var-lib-nfs-rpc_pipefs.mount
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:97]: if_error_exit?  if not, better comment.

File: tests/e2e/lib/diskutils [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+  umount /new_var
```
Comment (dougsland) [inline tests/e2e/lib/diskutils:98]: if_error_exit? or just warning? or just echo before ?

File: tests/e2e/set-ffi-env-e2e [hunk 12]
```diff
@@ -273,12 +244,30 @@ then
...
+
 echo
 info_message "Checking if QM already installed"
 info_message "=============================="
 QM_STATUS="$(systemctl is-enabled qm 2>&1)"
 if [ "$QM_STATUS" == "generated" ]; then
    if [ "$(systemctl is-active qm)" == "active" ]; then
+       # Restart QM after mount /var on separate partition
+       if grep -qi "${QC_SOC}" "${SOC_DISTRO_FILE}"; then
+          systemctl restart qm
```
Comment (dougsland) [inline tests/e2e/set-ffi-env-e2e:269]: if_error_exit?

**Review by dougsland — COMMENTED** [review]
LGTM, just small comments.

**Comment by dougsland** [comment]
The patch is okay to merge now, we can handle the rest in a next patch.

**Pull request merged**